liwords-ui/src/chat/chat.tsx

```diff
@@ -50,43 +85,142 @@ export const Chat = React.memo((props: Props) => {
     [props.presences]
   );
 
-  // Chat auto-scrolls when the last entity is visible.
-  const [hasUnreadChat, setHasUnreadChat] = useState(false);
-  const chatTab = selectedChatTab === 'CHAT' ? tabContainerElement : null;
-  const [chatAutoScroll, setChatAutoScroll] = useState(true);
   const doChatAutoScroll = useCallback(() => {
     if (chatAutoScroll && chatTab) {
-      const desiredScrollTop = chatTab.scrollHeight - chatTab.clientHeight;
       // Doing this conditionally may help browser performance.
       // (Not sure.)
-      if (chatTab.scrollTop !== desiredScrollTop) {
-        chatTab.scrollTop = desiredScrollTop;
+      if (chatTab.scrollTop !== chatTab.scrollHeight) {
+        chatTab.scrollTop = chatTab.scrollHeight;
       }
     }
   }, [chatAutoScroll, chatTab]);
 
+  const fetchChannels = useCallback(() => {
+    axios
+      .post(
+        toAPIUrl('user_service.SocializeService', 'GetActiveChatChannels'),
+        {
+          number: 20,
+          offset: 0,
+        },
+        { withCredentials: true }
+      )
+      .then((res) => {
+        const newChannels: ActiveChatChannels.AsObject = {
+          channelsList:
+            res.data.channels.map((ch: any) => {
+              return {
+                displayName: ch.display_name,
+                lastUpdate: parseInt(ch.last_update, 10),
+                hasUpdate: ch.has_update,
+                lastMessage: ch.last_message || '',
+                name: ch.name,
+              };
+            }) || [],
+        };
+        setChatChannels(newChannels);
+      });
+  }, [setChatChannels]);
```

Comment:
i believe that `setX` type state manipulation functions do not need to be specified as dependencies (they never change) and the linter picks up on this. I'll test it locally.

Reply:
Yeah, it seems pretty random as to when the linter yells at me or doesn't. I'll double check trying to remove these one by one.

Reply:
Nah. It wouldn't let me remove this one.

Reply:
if it's from a `useState` in the same function, it can be removed from dependency. since it's not, it needs to be specified as dependency.
https://github.com/domino14/liwords/blob/ea3587fe00d157a7246687c1b96a2ea7c1d33e39/liwords-ui/src/chat/chat.tsx#L62-L63